node/cmd/ccq/http.go

```diff
@@ -201,9 +202,16 @@ func (s *httpServer) handleQuery(w http.ResponseWriter, r *http.Request) {
 			return res.Signatures[i].Index < res.Signatures[j].Index
 		})
 		signatures := make([]string, 0, len(res.Signatures))
-		for _, s := range res.Signatures {
+		for _, sig := range res.Signatures {
+			if sig.Index > math.MaxUint8 {
+				s.logger.Error("Signature index out of bounds", zap.Int("sig.Index", sig.Index))
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				invalidQueryRequestReceived.WithLabelValues("failed_to_marshal_response").Inc()
+				failedQueriesByUser.WithLabelValues(permEntry.userName).Inc()
+				break
```

Comment:
this needs to break out of the `case` statement but it's in a for loop. won't it only break out of the loop and not the case?

e.g.
```go
package main

import (
	"fmt"
	"time"
)

func main() {
	select {
	case <-time.After(time.Second):
		for i := 0; i < 3; i++ {
			if i == 2 {
				break
			}
			fmt.Printf("i: %d\n", i)
		}
		fmt.Printf("still here")
	}
}
```
will print "still here"

Reply:
Thanks, fixed

Reply:
Yeah it looks kinda weird but apparently that's the Go way of doing it. Appreciate the review